src/ngraph/frontend/onnx_import/op/unsqueeze.cpp

```diff
@@ -33,7 +34,12 @@ namespace ngraph
                 {
                     auto data = node.get_ng_inputs().at(0);
                     auto axes = node.get_attribute_value<std::vector<std::int64_t>>("axes", {});
-                    const auto expanded_rank = data->get_shape().size() + axes.size();
+                    const auto data_rank = data->get_output_partial_shape(0).rank();
+                    CHECK_VALID_NODE(node,
+                                     data_rank.is_static(),
+                                     "Data rank must be static for calculate unsqueeze op");
```

Comment:
```suggestion
                                     "Data rank must be static for creation of ONNX Unsqueeze op");
```

Reply:
done

> Why not taking
> 
> https://github.com/NervanaSystems/ngraph/blob/8b9cdbb9a693731c103bd6f8d92fe59cf72e0a67/src/ngraph/validation_util.cpp#L851
> 
> `std::int64_t tensor_rank` this argument as for example: `std::uint32_t` or `std::uint64_t` type? IMHO this would be more safe.

Good point, I used `std::uint64_t` to `tensor_rank`, `axis_range_min`, `axis_range_max`